data/decks/waldstein.json

```diff
@@ -0,0 +1,201 @@
+{
+	"id": "waldstein",
+	"season": 6,
+	"display_name": "Waldstein",
+	"cardback": "cardback_waldstein.jpg",
+	"has_sealed_area": true,
+	"exceed_cost": 1,
+	"character_action_default": [{
+		"gauge_cost": 0,
+		"force_cost": 0,
+		"shortcut_effect_type": "strike",
+		"effect": {
+			"effect_type": "strike"
+		}
+	}],
+	"character_action_exceeded": [
+		{
+			"gauge_cost": 1,
+			"force_cost": 0,
+			"effect": {
+				"effect_type": "strike"
+			}
+		},
+		{
+			"gauge_cost": 0,
+			"force_cost": 0,
+			"min_hand_size": 1,
+			"action_name": "Revert",
+			"shortcut_effect_type": "gauge_from_hand",
+			"effect": {
+				"effect_type": "gauge_from_hand",
+				"min_amount": 1,
+				"max_amount": 1,
+				"and": {
+					"effect_type": "revert",
+					"and": {
+						"effect_type": "take_bonus_actions",
+						"amount": 1
+					}
+				}
+			}
+		}
+	],
+	"ability_effects": [
+		{
+			"character_effect": true,
+			"timing": "set_strike",
+			"condition": "used_character_action",
+			"condition_details": ["default", 0],
+			"effect_type": "add_attack_effect",
+			"added_effect": {
+				"timing": "hit",
+				"effect_type": "choice",
+				"choice": [
+					{ "effect_type": "push", "amount": 1 },
+					{ "effect_type": "pull", "amount": 1 }
+				]
+			}
+		}
+	],
+	"exceed_ability_effects": [
+		{
+			"character_effect": true,
+			"timing": "set_strike",
+			"condition": "used_character_action",
+			"condition_details": ["exceed", 0],
+			"effect_type": "powerup",
+			"amount": 2,
+			"and": {
+				"effect_type": "guardup",
+				"amount": 2,
+				"and": {
+					"timing": "cleanup",
```

Comment:
Just making a note as I go through, I don't expect this to work unless you've added new functionality.

Reply:
okay, after reviewing, I don't think I saw anything that would make this work?
No test for it either.
I think you need to have a separate exceed_ability_effect just for the timing: "cleanup"